doc/examples/transform/plot_register_rotation.py

```diff
@@ -0,0 +1,106 @@
+"""
+===================================
+Polar and Log-Polar Transformations
+===================================
+
+Rotation differences between two images can be converted to translation
+differences along the angular coordinate (:math:`\\theta`) axis of the
+polar-transformed images. Scaling differences can be converted to translation
+differences along the radial coordinate (:math:`\\rho`) axis if it
+is first log transformed (i.e., :math:`\\rho = \ln\sqrt{x^2 + y^2}`). Thus,
+in this example, we use phase correlation (``feature.register_translation``)
+to recover rotation and scaling differences between two images that share a
+center point.
```

Comment:
Nice example.

I would like to see this image as a fist plot: https://cloud.githubusercontent.com/assets/335370/22819610/26b2cce6-ef73-11e6-989c-997a8d44fb2b.png

This is rather simple to introduce the concept and will be a good thumbnail.
If you don't want to add it, I'll do later on.

Reply:
`data.retina()` is already in skimage.data.

Reply:
not entirely sure what you mean by "fist plot", but I agree that the retina image is more compelling for an example and have used it instead of the astronaut image. This also obviates the need for highlighting the area in the image to be transformed, because the image is naturally circular

Reply:
What I meant is just to reproduce the figure I linked in the previous message and display this figure in the first position. The two other figures that you made can come after.
The first figure is used as a thumbnail on the gallery. 

My point was to get something simple first, before showing some usage on the estimation of geometrical transforms.

But, this is just a suggestion. I'm happy with this PR anyway.

Reply:
Ahhhh, okay I understand now. Yes I agree that could be useful. The information is more or less there in the two plots I have, but those aren't really optimized for demonstrating the basic concept of the transform. I will add later, or you can feel free to drop it in.